tw-tkms-starter/src/main/java/com/transferwise/kafka/tkms/TkmsStorageToKafkaProxy.java

```diff
@@ -372,6 +392,10 @@ private void poll(Control control, TkmsShardPartition shardPartition) {
                 if (failedSendsCount.get() > 0) {
                   proxyCyclePauseRequest.setValue(tkmsPaceMaker.getPollingPauseOnError(shardPartition));
                 }
+              } catch (InterruptException e) {
+                log.error("Kafka producer was interrupted for " + shardPartition + ".", e);
+                // Rethrow and force the recreation of the producer.
+                throw e;
```

Comment:
Why do we handle this case differently from the one below?

Reply:
Yep, we are logging down a trackable message.

And we force the poll loop to exit, and thus, a new producer instance to be created.